plugin/output/s3/s3.go

```diff
@@ -262,3 +290,122 @@ func (p *Plugin) generateObjectName(name string) string {
 	objectName = objectName[0 : len(objectName)-len(p.compressor.getExtension())]
 	return fmt.Sprintf("%s.%s%s", objectName, n, p.compressor.getExtension())
 }
+
+func (p *Plugin) createMinioClients() (minioClients map[string]objectStoreClient) {
+	minioClients = make(map[string]objectStoreClient)
+	// initialize minio clients object for main bucket.
+	mainClient, err := minio.New(p.config.Endpoint, p.config.AccessKey, p.config.SecretKey, p.config.Secure)
+	if err != nil || mainClient == nil {
+		p.logger.Panicf("could not create minio client, error: %s", err.Error())
+	}
+	minioClients[p.config.Bucket] = mainClient
+
+	// TODO create minioClients on upper level, dependency inversion will be fixed, mock will be set in test, not in Config.client.
+	if p.config.client != nil {
+		p.logger.Info("set mock clients")
+		minioClients[p.config.Bucket] = *p.config.client
+	}
+
+	for _, singleB := range p.config.MultiBuckets {
+		bClient, err := minio.New(singleB.Endpoint, singleB.AccessKey, singleB.SecretKey, singleB.Secure)
+		if err != nil || mainClient == nil {
+			p.logger.Panicf("could not create minio client, error: %s", err.Error())
+		}
+		minioClients[singleB.Bucket] = bClient
+
+		// TODO create minioClients on upper level, dependency inversion will be fixed, mock will be set in test, not in Config.client.
+		if p.config.client != nil {
+			p.logger.Info("set mock clients")
+			minioClients[singleB.Bucket] = *p.config.client
+		}
+	}
+
+	return minioClients
+}
+
+func (p *Plugin) setDirs(outPlugCount int) {
+	// dir for all bucket files.
+	dir, _ := filepath.Split(p.config.FileConfig.TargetFile)
+
+	targetDirs := make(map[string]string, outPlugCount)
+	targetDirs[p.config.Bucket] = dir
+	for _, singleB := range p.config.MultiBuckets {
+		targetDirs[singleB.Bucket] = fmt.Sprintf("%s%s/", dir, singleB.Bucket)
```

Comment:
it is better to use filepath.Join in compressFilesInDir and here "/" will not be needed.

Reply:
Thx, I added this idea, but we need need use "/"